ouroboros-consensus/src/Ouroboros/Consensus/Storage/ImmutableDB/Impl.hs

```diff
@@ -212,31 +212,40 @@ deleteAfter = deleteAfter_
 ------------------------------------------------------------------------------}
 
 openDB ::
-     forall m blk.
+     forall m blk ans.
      ( IOLike m
      , GetPrevHash blk
      , ConvertRawHash blk
      , ImmutableDbSerialiseConstraints blk
      , HasCallStack
      )
   => ImmutableDbArgs Identity m blk
-  -> m (ImmutableDB m blk)
-openDB args = fst <$> openDBInternal args
+  -> (
+       (forall h.
+        WithTempRegistry (OpenState m blk h) m (ImmutableDB m blk, OpenState m blk h, OpenState m blk h, OpenState m blk h -> m Bool)
```

Comment:
I pushed up a commit on `nfrisby/CAD-3565-split-openChainDB` that I think simplifies this. @Jasagredo What do you think?

Reply:
Looks super good to me :+1: